cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -416,6 +420,79 @@ func (s *Supervisor) startOpAMP() error {
 	return nil
 }
 
+func (s *Supervisor) stopOpAMP() error {
+	s.logger.Debug("Stopping OpAMP client...")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	err := s.opampClient.Stop(ctx)
+	// TODO(srikanthccv): remove context.DeadlineExceeded after https://github.com/open-telemetry/opamp-go/pull/213
+	if err != nil && !errors.Is(err, context.DeadlineExceeded) {
+		return err
+	}
+	s.logger.Debug("OpAMP client stopped.")
+	return nil
+}
+
+func (s *Supervisor) getHeadersFromSettings(protoHeaders *protobufs.Headers) http.Header {
+	headers := make(http.Header)
+	for _, header := range protoHeaders.Headers {
+		headers.Add(header.Key, header.Value)
+	}
+	return headers
+}
+
+func (s *Supervisor) onOpampConnectionSettings(_ context.Context, settings *protobufs.OpAMPConnectionSettings) error {
+	if settings == nil {
+		s.logger.Debug("Received ConnectionSettings request with nil settings")
+		return nil
+	}
+
+	newServerConfig := &config.OpAMPServer{}
+
+	if settings.DestinationEndpoint != "" {
```

Comment:
I wonder if there any value in using some of the `confmap` work to merge these values together instead of needing to update each one explicitly?

Reply:
I am not very familiar with `confmap` work. I will take a look at it and If it helps here I can send a separate PR for it.